src/executionFiles/StepExecutor.ts

```diff
@@ -52,53 +52,28 @@ export class StepExecutor {
 
   executeStep = async (signer: Signer, step: Step): Promise<Step> => {
     // check if signer is for correct chain
-    if ((await signer.getChainId()) !== step.action.fromChainId) {
-      // -> set status message
-      step.execution = this.statusManager.initExecutionObject(step)
-      this.statusManager.updateExecution(step, 'CHAIN_SWITCH_REQUIRED')
-      const chain = getChainById(step.action.fromChainId)
-
-      const switchProcess = this.statusManager.findOrCreateProcess(
-        'switchProcess',
-        step,
-        `Change Chain to ${chain.name}`
-      )
-
-      if (this.executionStopped) return step
-
-      let updatedSigner
-      try {
-        updatedSigner = await this.settings.switchChainHook(
-          step.action.fromChainId
-        )
-        if (
-          updatedSigner &&
-          (await updatedSigner.getChainId()) === step.action.fromChainId
-        ) {
-          signer = updatedSigner
-        } else {
-          throw Error('CHAIN SWITCH REQUIRED')
-        }
-      } catch (e: any) {
-        this.statusManager.updateProcess(step, switchProcess.id, 'FAILED', {
-          errorMessage: e.message,
-          errorCode: e.code,
-        })
-        this.statusManager.updateExecution(step, 'FAILED')
-        throw e
-      }
+    const updatedSigner = await switchChain(
```

Comment:
shouldn't that only occur in the stepexecutors, when its really needed?

Reply:
this might be scope creep, as it is another issue: https://lifi.atlassian.net/browse/LF-126

Reply:
Yeah I would agree that thats a different feature. My implementation should not change the behavior here, just extract the code. Let's deal with that problem in the other ticket